hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,164 @@
+
+export class HelloWorldSemanticTokenProvider extends AbstractSemanticTokenProvider {
+    protected override highlightElement(node: AstNode, acceptor: SemanticTokenAcceptor): void | "prune" | undefined {
+        if (isPerson(node)) {
+            acceptor({
+                node,
+                property: 'name',
+                type: SemanticTokenTypes.class
+            });
+        }
+    }
+}
+```
+
+For all persons (`isPerson(...)` in line 7), we explicitly specify the semantic type for the token of their `'name'` property.
+Here, we use `SemanticTokenTypes.class` as semantic type.
+For your case, select a predefined type which fits your domain best.
+Since the name is used as cross-reference by greetings, a similar check and assignment of a semantic token type needs to be done for the `person` property of `Greeting` as well.
+
+After creating the semantic token provider, you need to register the `HelloWorldSemanticTokenProvider` in `hello-world-module.ts` in the following way:
+
+```ts
+export const HelloWorldModule: Module<HelloWorldServices, PartialLangiumServices & HelloWorldAddedServices> = {
+    lsp: {
+        SemanticTokenProvider: (services) => new HelloWorldSemanticTokenProvider(services)
+    },
+    validation: {
+        HelloWorldValidator: () => new HelloWorldValidator()
+    }
+};
+```
+
+Now rebuild and restart your application and test the improvements of the second step:
+
+![screenshot with fixed semantic token](fixed-2-token.png)
+
+The `HelloWorldSemanticTokenProvider` works, and you might see a different highlighting XOR you might not see any difference, e.g. "Hello" is still blue here. This leads us to the third step.
+
+
+## Step 3: Ensure that your editor styles the chosen semantic token type
+
+The __third step__ is to ensure, that your editor supports the assigned semantic tokens:
+Depending on your editor and the currently selected color theme, the semantic token type selected in `HelloWorldSemanticTokenProvider` might not be supported or didn't got a different color in the color theme.
+The easiest way to detect such problems is to change the current color theme and to try some others.
+Note, that VS Code allows to switch off semantic highlighting for all themes with the setting `editor.semanticHighlighting.enabled`.
+
+After switching from "Dark (Visual Studio)" to "Dark Modern" in VS Code, the example looks as expected.
+You can switch the current color theme in VS Code with `cmd + K` `cmd + T` (or via the menu: Code -> Settings... -> Theme -> Color Theme).
+
+![screenshot with supporting color theme](fixed-3-style-2.png)
+
+Now "Hello" is highlighted in purple, when it is used as keyword, and in green, when it is used as value for the name of a person.
+Another solution is to select a different semantic type for your token in step two.
+A more elaborate solution is to create your own color theme and to ship it with your VS Code extension of your language.
```

Comment:
I would not mention that here as users would have to pro-actively select the specific color theme, wouldn't they?

Reply:
You are right, that seems to be not supported by VS Code according to https://github.com/Microsoft/vscode/issues/20652.